streams/readable-streams/floating-point-total-queue-size.js

```diff
@@ -0,0 +1,86 @@
+'use strict';
+
+if (self.importScripts) {
+  self.importScripts('/resources/testharness.js');
+}
+
+promise_test(() => {
+  const { reader, controller } = setupTestStream();
+
+  controller.enqueue(2);
+  assert_equals(controller.desiredSize, 0 - 2, 'desiredSize must be -2 after enqueueing such a chunk');
+
+  controller.enqueue(Number.MAX_SAFE_INTEGER);
+  assert_equals(controller.desiredSize, 0 - Number.MAX_SAFE_INTEGER - 2,
+    'desiredSize must be calculated using floating-point arithmetic (adding a second chunk)');
+
+  return reader.read().then(() => {
+    assert_equals(controller.desiredSize, 0 - Number.MAX_SAFE_INTEGER - 2 + 2,
+      'desiredSize must be calculated using floating-point arithmetic (subtracting a chunk)');
+
+    return reader.read();
+  }).then(() => {
+    assert_equals(controller.desiredSize, 0 - Number.MAX_SAFE_INTEGER - 2 + 2 + Number.MAX_SAFE_INTEGER,
```

Comment:
Failing expectation 2 from https://github.com/whatwg/streams/issues/582#issuecomment-273702677 like this when the queue is empty still seems really weird, when ResetQueue could be easily used for that case too rather than letting [[queueTotalSize]] e.g. become negative / mess with backpressure expectations.